src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,22 +55,53 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block + ", " + street + ", " + unit + ", " + postalCode;
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        if (other == this) // short circuit if same object
+            return true;
+        
+        if (other instanceof Address) { // instanceof handles nulls
+            Address otherAddress = (Address)other;
+            return otherAddress.block.equals(block) &&
+                    otherAddress.street.equals(street) &&
+                    otherAddress.unit.equals(unit) &&
+                    otherAddress.postalCode.equals(postalCode);
+        }
+        
+        return false;
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        int prime = 31;
```

Comment:
you could use `Object.hash` that is found elsewhere in the codebase (please search for it as I don't recollect exactly); and explicitly take note of the comment associated with that ;-)